.github/CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,46 @@
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, nationality, personal appearance, race, religion, or sexual identity and orientation.
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment include:
+
+* Using welcoming and inclusive language
+* Being respectful of differing viewpoints and experiences
+* Gracefully accepting constructive criticism
+* Focusing on what is best for the community
+* Showing empathy towards other community members
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery and unwelcome sexual attention or advances
+* Trolling, insulting/derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or electronic address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a professional setting
+
+## Our Responsibilities
+
+Project maintainers are responsible for clarifying the standards of acceptable behavior and are expected to take appropriate and fair corrective action in response to any instances of unacceptable behavior.
+
+Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, or to ban temporarily or permanently any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful.
+
+## Scope
+
+This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at coc@symfony.com. The project team will review and investigate all complaints, and will respond in a way that it deems appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.
+
+Project maintainers who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent repercussions as determined by other members of the project's leadership.
```

Comment:
We should reword this --> ` as determined by other members of the project's leadership` to be explicit. Let's list the offends and corresponding punishments.

Reply:
in general imho we should bring such feedback upstream. I would really prefer not forking the contents of this standard CoC.

Reply:
BTW I think on this aspect .. the CoC tries to be specific to some degree on the offenses .. but if we were to attempt to list the offenses and punishments exhaustively .. I think we would need to fill a book, which I think isn't really helpful.

Reply:
Yes, I agree we can't list them all. We have two alternatives:

1. Saying that you can't do A, B, C, D and E.
2. Saying that you can't do A, B, C, D and E ... and whatever any other thing a certain group of appointed people think you can't do.

I can't accept **2.**. That's why I propose to use **1.** and remove any vague statement that is a backdoor to power abuse.

Reply:
I understand where you are coming from but I think your expectations are unrealistic. Again if we try to list exactly every little thing, then we will miss things which enables predators and as a result the entire thing is of little value. Even the law doesn’t have such details and leaves some things up to judges and I don’t think we want to maintain anything even remotely close to the rule of law.

Reply:
@lsmith77 thanks for bringing in the Law to this discussion. I think your comment is not correct because the Law is the opposite of what you said. Specifically, this principle: *"Everything which is not forbidden is allowed"* ([explanation on Wikipedia](https://en.wikipedia.org/wiki/Everything_which_is_not_forbidden_is_allowed)).

Can you imagine a country where the Law says: *"you can't do A, B, C, D, E ... and whatever the elected President decides you can't do"*. That's not how the Law works.

About the judges, again I think your comment is not entirely true. The "Criminal Code" of each country lists the offences  and their punishments exhaustively. If it's not in the "Criminal Code", then it's not a crime. Judges can adjust the punishments ... but always within the limits defined by the Criminal Code for each crime.

Reply:
exactly .. and I am saying you cannot expect a CoC to work like criminal law. at the end of the day we need to ensure that minorities feel safe in our community. So yes there will be edge cases and those will be hard to deal with simply because we do not have the infrastructure that is used to enforce the law. That being said, when you say “I cannot accept 2.” .. this is effectively what we have today. Right now nothing is written down, yet I assure you that even without a CoC community members could be expelled if they misbehave. This CoC is already a step in the direction of 1. that you hope for ..

Reply:
This is where we entirely disagree. To me, the two most important things a CoC must do are:

1. List the things you can't do in this community.
2. Explain what happens to you when you do those things.

The list of forbidden things will never be perfect, but can be modified in time. If we get some report about something that it's not on that list but we consider a wrong thing, then we cannot punish that person (because it wasn't on the list) ... but we can add it to the list to prevent that from happening in the future.

Reply:
@javiereguiluz Laws are often much less concrete than you implied. For example, the [US Fair Debt Collection Practices Act](https://www.ftc.gov/enforcement/rules/rulemaking-regulatory-reform-proceedings/fair-debt-collection-practices-act-text) includes the following language:

> See, e.g., 15 U.S.C. §§ 1692c(a)(1) (placing restrictions on call times), 1692d(5) (prohibiting “any conduct the natural consequence of which is to harass, oppress, or abuse,” including “[c]ausing a telephone to ring or engaging any person in telephone conversation repeatedly or continuously with intent to annoy, abuse, or harass”).

The law does not define "harass, oppress, or abuse", "repeatedly", or "continuously". Instead, it is the explicit purpose of a court to interpret the law, the situation, and issue a remedy. Regardless, perhaps the courts aren't the example we should be shooting for, as they [regularly](https://www.techdirt.com/articles/20100305/0404088432.shtml) [stretch](https://www.reuters.com/article/us-myanmar-lawyers/myanmars-courts-stretch-laws-and-credibility-idUSBKK9180620080313) [the](http://www.bscr-law.com/?t=40&an=36062&stylesheet=blog&p=5258) [law](https://digitalcommons.ilr.cornell.edu/cgi/viewcontent.cgi?article=2035&context=articles), as well.

Having said that, there must be a balance between "explicit" and "implied" that can be agreed upon; perhaps even a result that can be merged upstream.